### PR TITLE
Frontier/js: ethereum.js in CLI

### PR DESCRIPTION
**Pull request by zelig**
With this PR, the DAPP JS API implemented by https://github.com/ethereum/ethereum.js is exposed in the Ethereum Frontier Javascript environment (VM and interactive console).
rpc/jeth: It uses a native binding in the otto javascript VM to implement a 'provider' (RPC transport layer) between JS and ETH
rpc message fields no longer use camelcase, but titlecase 
jsre: package provides an eth-less JSRE wrapper around otto VM
see https://github.com/ethereum/go-ethereum/issues/459

### CODE REVIEW

**Comment by obscuren** [comment]
Fix the tests please

File: cmd/mist/assets/qml/main.qml [hunk 1]
```diff
@@ -158,105 +158,6 @@ ApplicationWindow {
 
 
 
-    menuBar: MenuBar {
```
Comment (obscuren) [inline cmd/mist/assets/qml/main.qml]: Why did you remove this? Did you forget to `git add .` maybe?

File: cmd/ethereum/main.go [hunk 2]
```diff
@@ -89,16 +89,20 @@ Use "ethereum dump 0" to dump the genesis block.
 `,
 		},
 		{
-			Action: runjs,
+			Action: console,
+			Name:   "console",
+			Usage:  `Ethereum Frontier Console: interactive JavaScript environment`,
+			Description: `
+Frontier Console is an interactive shell for the Ethereum Frontier JavaScript runtime environment which exposes a node admin interface as well as the DAPP JavaScript API.
```
Comment (obscuren) [inline cmd/ethereum/main.go]: It isn't specifically for frontier. Remove references to **Frontier**

Comment (zelig) [inline cmd/ethereum/main.go]: https://github.com/ethereum/go-ethereum/wiki/Frontier-Console
